src/Essentials/src/DeviceDisplay/DeviceDisplay.ios.cs

```diff
@@ -1,20 +1,108 @@
 #nullable enable
 using System;
+using System.Runtime.InteropServices;
 using Foundation;
 using UIKit;
+using ObjCRuntime;
 
 namespace Microsoft.Maui.Devices
 {
 	partial class DeviceDisplayImplementation : IDeviceDisplay
 	{
 		NSObject? observer;
 
+#if MACCATALYST
+		const string NSApplicationDidChangeScreenParametersNotification = "NSApplicationDidChangeScreenParametersNotification";
+#endif
+
+#if MACCATALYST
+		// Core Graphics P/Invoke declarations for Mac Catalyst
+		// Returns the display ID of the main display
+		[DllImport(Constants.CoreGraphicsLibrary)]
+		static extern uint CGMainDisplayID();
```

Comment:
`CGMainDisplayID()` only returns the main display, ignoring external monitors. I think that if user drags app window to external monitor and changes that monitor's refresh rate/resolution, the event fires but reports the main display's properties, not the active display's.